lib/mongo/crypt/auto_encrypter.rb

```diff
@@ -54,21 +54,56 @@ def setup_encrypter(options = {})
         extra_options = options.delete(:extra_options)
         extra_options = DEFAULT_EXTRA_OPTIONS.merge(extra_options)
 
-        mongocryptd_client_monitoring_io = extra_options.delete(:mongocryptd_client_monitoring_io)
+        monitoring_io_option = extra_options.delete(:mongocryptd_client_monitoring_io)
+
+        if monitoring_io_option.nil?
+          mongocryptd_client_monitoring_io = true
+        else
+          mongocryptd_client_monitoring_io = monitoring_io_option
+        end
 
         opts_copy = options.dup
         opts_copy[:bypass_auto_encryption] = opts_copy[:bypass_auto_encryption] || false
 
         super(opts_copy.merge(extra_options))
 
+        # validate_extra_options!
+
         @mongocryptd_client = Client.new(
                                 @encryption_options[:mongocryptd_uri],
                                 monitoring_io: mongocryptd_client_monitoring_io,
                               )
 
+        if !@encryption_options[:mongocryptd_bypass_spawn] && mongocryptd_client_monitoring_io
+          self.spawn_mongocryptd
+        end
+
         # TODO: use all the other options for auto-encryption/auto-decryption
       end
 
+      # Spawn a new mongocryptd process using the mongocryptd_spawn_path
+      # and mongocryptd_spawn_args passed in through the extra auto
+      # encrypt options. Stdout and Stderr of this new process are written
+      # to /dev/null.
+      #
+      # @note To capture the mongocryptd logs, add "--logpath=/path/to/logs"
+      #   to auto_encryption_options -> extra_options -> mongocrpytd_spawn_args
+      #
+      # @return [ Integer ] The process id of the spawned process
+      def spawn_mongocryptd
+        mongocryptd_spawn_args = @encryption_options[:mongocryptd_spawn_args]
+
+        if  mongocryptd_spawn_args.nil? || mongocryptd_spawn_args.empty?
+          mongocryptd_spawn_args = ['--idleShutdownTimeoutSecs=0']
```

Comment:
@p-mongo -- Thank you for your comment regarding `Process.spawn` executing any command if the list of arguments is empty. I came up with two potential ways of solving this problem:

1. Always pass an argument to the mongocryptd process. By default, mongocryptd sets the `idleShutdownTimeoutSecs` argument to 0, so there is no harm (currently) in passing that in as a default. If the user passes in a path that isn't an executable, `Process.spawn` will error. Of course, this could break something if server team changes how mongocryptd works in later versions of the server.

2. (Try to) validate that the spawn path is the path to an executable. I wrote a method below called `validate_extra_options!`, which checks that the path is an executable file or that it is referencing an executable in the user's PATH variable. There is a similar method already written in the mkmf module ([reference here](https://ruby-doc.org/stdlib-2.0.0/libdoc/mkmf/rdoc/MakeMakefile.html#method-i-find_executable)) but that method writes to stdout, and it would require adding another dependency to the project.

Let me know what you think!

Reply:
Will follow up as https://jira.mongodb.org/browse/RUBY-2034.